heat/core/dndarray.py

```diff
@@ -1769,12 +1792,16 @@ def is_balanced(self):
         balanced : bool
             True if balanced, False if not
         """
+        if self.balanced is not None:
+            return self.balanced
```

Comment:
is it worth doing the check here? perhaps this function should actually check every time. if we want to avoid the check i think we would use `if a.balanced:` instead of `if a.is_balanced()`. although this seems like its splitting hairs

Reply:
I started out like you say, and then I found myself writing `if a.balanced is None, do a.is_balanced()` too many times.

I do agree though that there should be a way to force the check. I was thinking `a.is_balanced(force=True)`

Reply:
maybe check instead of force? or better yet, `force_check`?

Reply:
Yes, `force_check` sounds perfect